banking/overrides/bank_transaction.py

```diff
@@ -72,3 +76,251 @@ def on_update_after_submit(doc, event):
 				),
 				title=_("Over Allocation"),
 			)
+
+
+def before_validate(doc, method):
+	ensure_positive_deposit_withdrawal_fees(doc, method)
+
+
+def on_cancel(doc, method):
+	# Cancel the journal entries created by this Bank Transaction
+	auto_created_journal_entries = frappe.get_all(
+		"Journal Entry",
+		filters={"cheque_no": doc.name},
+		pluck="name",
+	)
+
+	for journal_entry in auto_created_journal_entries:
+		try:
+			doc = frappe.get_doc("Journal Entry", journal_entry)
+			if doc.docstatus == 1:
+				doc.cancel()
+		except Exception as e:
+			frappe.msgprint(f"Failed to cancel {journal_entry}: {e}")
```

Comment:
```suggestion
			frappe.msgprint(_("Failed to cancel {0}: {1}").format(journal_entry, e))
```
That is how it is translatable. Note: The syntax of the suggestion might be a bit off. Please, check it again.